pkg/virt-handler/vm.go

```diff
@@ -447,6 +501,21 @@ func (d *VirtualMachineController) updateVMIStatus(vmi *v1.VirtualMachineInstanc
 					// Only interfaces defined in domain.Spec are handled here
 					newInterface = existingInterface
 					newInterface.MAC = interfaceMAC
+
+					// If it is a Combination of Masquerade+Pod network, check IP from file cache
+					if existingInterfacesSpecByName[domainInterface.Alias.Name].Masquerade != nil && existingNetworksByName[domainInterface.Alias.Name].NetworkSource.Pod != nil {
+						iface, err := d.getPodInterfacefromFileCache(vmi, domainInterface.Alias.Name)
+						if err != nil {
+							return err
+						}
+						if iface.PodIP != existingInterfaceStatusByName[domainInterface.Alias.Name].IP {
+							newInterface = v1.VirtualMachineInstanceNetworkInterface{
```

Comment:
* Why is a new v1.VirtualMachineInstanceNetworkInterface being created instead of reusing the existing one? Although it doesn't seem any existing fields are lost/overriden. For consistency with the [rest of the code](https://github.com/kubevirt/kubevirt/pull/3488/files#diff-27207346391eaa19190c76760eb8851fR530-R535) I would just update the 'IP' and 'IPs' fields on the existing instance.
* Shouldn't the 'IPs' field contain the pod ip as well?

Reply:
Corrected now